lib/pbench/server/filetree.py

```diff
@@ -339,68 +338,69 @@ def unpack(self, incoming: Path, results: Path):
+                process = subprocess.run(
+                    [
+                        "tar",
+                        "-x",
+                        "--no-same-owner",
+                        "--delay-directory-restore",
+                        "--force-local",
+                        "--file=" + str(self.tarball_path),
+                    ],
+                    cwd=unpacked,
+                    stdin=subprocess.DEVNULL,
+                    capture_output=True,
+                    text=True,
                 )
-
-        find_command = "find . ( -type d -exec chmod ugo+rx {} + ) -o ( -type f -exec chmod ugo+r {} + )"
-        try:
-            process = subprocess.run(
-                find_command.split(),
-                cwd=unpacked,
-                stdin=subprocess.DEVNULL,
-                capture_output=True,
-                text=True,
-            )
-        except Exception as exc:
-            raise TarballModeChangeError(unpacked, str(exc)) from exc
-        else:
-            if process.returncode != 0:
-                raise TarballModeChangeError(
-                    unpacked,
-                    f"tar exited with status {process.returncode}:  {process.stderr.strip()!r}",
+            except Exception as exc:
+                raise TarballUnpackError(self.tarball_path, str(exc)) from exc
+            else:
+                if process.returncode != 0:
+                    raise TarballUnpackError(
+                        self.tarball_path,
+                        f"tar exited with status {process.returncode}:  {process.stderr.strip()!r}",
+                    )
+
+            find_command = "find . ( -type d -exec chmod ugo+rx {} + ) -o ( -type f -exec chmod ugo+r {} + )"
+            try:
+                process = subprocess.run(
+                    find_command.split(),
+                    cwd=unpacked,
+                    stdin=subprocess.DEVNULL,
+                    capture_output=True,
+                    text=True,
                 )
+            except Exception as exc:
+                raise TarballModeChangeError(unpacked, str(exc)) from exc
+            else:
+                if process.returncode != 0:
+                    raise TarballModeChangeError(
+                        unpacked,
+                        f"tar exited with status {process.returncode}:  {process.stderr.strip()!r}",
+                    )
 
-        try:
-            shutil.move(unpacked / self.name, incoming)
-        except Exception as exc:
             try:
-                shutil.rmtree(incoming)
-            except Exception as e:
+                shutil.move(unpacked / self.name, incoming)
+            except Exception as exc:
+                try:
+                    shutil.rmtree(incoming / self.name, ignore_errors=True)
+                except Exception as e:
+                    self.logger.error(
+                        f"Error in removing partially copied Tarball: {str(e)}"
+                    )
```

Comment:
You don't need this, anymore. With the `finally` we'll always delete the `incoming / self.name` directory in one place. The `ignore_errors=True` here, however, should be moved down to that `rmtree`.